tests/data/menu_example/menu_example.module

```diff
@@ -377,7 +377,7 @@ function menu_example_permission() {
  *   True if the acting user has the role specified.
  */
 function menu_example_custom_access($role_name) {
-  $access_granted = in_array($role_name, $GLOBALS['user']->roles);
+  $access_granted = array_key_exists($role_name, $GLOBALS['user']->roles);
   return $access_granted;
```

Comment:
If this didn't work, we need to fix the `GlobalUser` and how it returns `roles` as this was the previous code from Drupal 7.

Reply:
In D7, roles is keyed by numeric role ID. In D10, roles is keyed by role name and the values are role objects.